src/test/java/org/javarosa/core/model/test/FormDefTest.java

```diff
@@ -173,6 +174,7 @@ public void repeatIsIrrelevant_whenGrandparentRelevanceSetToFalse() throws IOExc
     }
 
     @Test
+    @Ignore("Test needs rewritten")
```

Comment:
This test could possibly have helped us notice the bug here. The relevance of the inner group is based on the position of the **outer group** but I think I convinced myself it was based on the position of the inner group (and that's what I was thinking when we talked, @seadowg).

In the original test, the first assertion is:
```
scenario.next();
scenario.next();
assertThat(scenario.refAtIndex(), is(getRef("/data/outer[0]/inner[1]")));
```

This is wrong because of the bug this PR fixes. Any inner group should be non-relevant based on the outer group's index. The form definition instance only has one instance of the inner repeat. It turns out the event emitted is actually a repeat prompt (because of this bug). The rest of the test makes sense.

Below is a more explicit proposal. I put two instances of the outer repeat and two inner instances in the first outer instance. That way we can check that initially none of the inner repeats in the first outer repeat are relevant and that there's no prompt. 

```java
Test
public void nestedRepeatRelevance_updatesBasedOnParentPosition() throws IOException {
    Scenario scenario = Scenario.init("Nested repeat relevance", html(
        head(
            title("Nested repeat relevance"),
            model(
                mainInstance(t("data id=\"nested-repeat-relevance\"",
                    t("outer",
                        t("inner",
                            t("q1")
                        ),
                        t("inner",
                            t("q1")
                        )
                    ),
                    t("outer",
                        t("inner",
                            t("q1")
                        )
                    ),
                    t("relevance-condition", "0")
                )),
                bind("/data/relevance-condition").type("string"),
                bind("/data/outer/inner").relevant("position(..) mod 2 = /data/relevance-condition")
            ),
            body(
                repeat("/data/outer",
                    repeat("/data/outer/inner",
                        input("/data/outer/inner/q1")
                    )
                ),
                input("/data/relevance-condition")
            ))));

    FormDef formDef = scenario.getFormDef();
    FormEntryModel formEntryModel = new FormEntryModel(formDef);

    // For ref /data/outer[0]/inner[0], the parent position is 1 so the boolean expression is false. That means
    // none of the inner groups in /data/outer[0] can be relevant.
    assertThat(formEntryModel.isIndexRelevant(scenario.indexOf("/data/outer[0]/inner[0]")), is(false));

    scenario.next();
    // The repeat is relevant but nothing inside it is.
    assertThat(scenario.refAtIndex(), is(getRef("/data/outer[0]")));

    scenario.next();
    assertThat(scenario.refAtIndex(), is(getRef("/data/outer[1]")));

    scenario.next();
    assertThat(scenario.refAtIndex(), is(getRef("/data/outer[1]/inner[0]")));

    scenario.next();
    assertThat(scenario.refAtIndex(), is(getRef("/data/outer[1]/inner[0]/q1[0]")));

    scenario.answer("/data/relevance-condition", "1");

    scenario.jumpToBeginningOfForm();
    scenario.next();
    assertThat(scenario.refAtIndex(), is(getRef("/data/outer[0]")));

    scenario.next();
    assertThat(scenario.refAtIndex(), is(getRef("/data/outer[0]/inner[0]")));
}
```

Reply:
Right this makes a lot more sense to me now. The way we were thinking about position before was hurting my brain. I think we can incorporate this in this PR.